### PR TITLE
feat: implement NFTJson Metadata

### PR DESCRIPTION
**Pull request by Iwueseiter**
- issue #181
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/art-peace/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: backend/routes/indexer/nft.go [hunk 1]
```diff
@@ -135,6 +136,36 @@ func processNFTMintedEvent(event IndexerEvent) {
 		return
 	}
 
+	// Create a NFT JSON metadata file
+	metadata := map[string]interface{}{
+		"name":        fmt.Sprintf("NFT #%d", tokenId),
+		"description": "This is an NFT minted on the blockchain.",
+		"image":       fmt.Sprintf("ipfs://%s", imageHashHex),
```
Comment (b-j-roberts) [inline backend/routes/indexer/nft.go:143]: Image links will be to our backend for now.

File: backend/routes/indexer/nft.go [hunk 2]
```diff
@@ -135,6 +136,36 @@ func processNFTMintedEvent(event IndexerEvent) {
...
+		"attributes": []map[string]string{
+			{
+				"trait_type": "Width",
+				"value":      fmt.Sprintf("%d", width),
+			},
+			{
+				"trait_type": "Height",
+				"value":      fmt.Sprintf("%d", height),
+			},
+		},
```
Comment (b-j-roberts) [inline backend/routes/indexer/nft.go:153]: An attribute for Position ( x, y ) would be cool, also the minter's name or address.

**Review by b-j-roberts — CHANGES_REQUESTED** [review]
Nice work so far!

File: backend/routes/indexer/nft.go (hunk 1 above)
Comment (Iwueseiter) [reply backend/routes/indexer/nft.go:143]: @b-j-roberts please can you explain better what you want here?

Comment (b-j-roberts) [reply backend/routes/indexer/nft.go:143]: This field stores a url to the image we generate of the NFT, which is exposed on the backend through the `nft-images/nft-{token id}.png` route. We wont be using ipfs to store images, so the ipfs url is incorrect.

**Comment by b-j-roberts** [comment]
Thank you, nice work!

**Pull request merged**